docs/posts/posts.data.ts

```diff
@@ -0,0 +1,3 @@
+import { createContentLoader } from 'vitepress'
+
+export default createContentLoader('posts/*.md', )
```

Comment:
This file obtains the metadata from the posts and passes it to the index, this implementation is a very simple builtin provided by VitePress and is sufficient for this first milestone.

If we wanted to read the posts from the main branch with the build from the latest branch, this implementation would need to be modified to use the GitHub API, for now and for simplicity, I've decided to leave it as is.